docker/Dockerfile.rhel8

```
@@ -0,0 +1,112 @@
+RUN mkdir -p /etc/sudoers.d
+RUN touch /etc/sudoers.d/opam 
+RUN chmod 440 /etc/sudoers.d/opam 
+RUN chown root:root /etc/sudoers.d/opam 
+RUN adduser --uid 1000 --password '' opam 
+RUN usermod -aG wheel opam
+RUN passwd -d opam 
+RUN chown -R opam:opam /home/opam 
+USER opam
+ENV HOME=/home/opam
+WORKDIR /home/opam
+RUN mkdir .ssh 
+RUN chmod 700 .ssh 
+RUN touch /home/opam/.opamrc-nosandbox 
+RUN touch /home/opam/opam-sandbox-disable 
+RUN chmod a+x /home/opam/opam-sandbox-disable 
+RUN sudo mv /home/opam/opam-sandbox-disable /usr/bin/opam-sandbox-disable 
+RUN touch /home/opam/.opamrc-sandbox 
+RUN touch /home/opam/opam-sandbox-enable 
+RUN chmod a+x /home/opam/opam-sandbox-enable 
+RUN sudo mv /home/opam/opam-sandbox-enable /usr/bin/opam-sandbox-enable 
+RUN git config --global user.email "docker@example.com" 
+RUN git config --global user.name "Docker" 
+RUN opam-sandbox-disable 
+COPY --from=opam /home/opam/opam-repository /home/opam/opam-repository
+RUN opam init -k local -a /home/opam/opam-repository --bare --disable-sandboxing
+RUN rm -rf .opam/repo/default/.git 
+ENV OPAMYES=1 OPAMCONFIRMLEVEL=unsafe-yes OPAMERRLOGLEN=0 OPAMPRECISETRACKING=1
+RUN opam switch create 4.12 --packages=ocaml-base-compiler.4.12.1 
+RUN opam update 
+RUN opam pin add -k version ocaml-base-compiler 4.12.1 
+RUN opam install -y opam-depext 
+
+# we really don't want to build from source, why does 'yum install libev-devel' not work?
+# these dependencies are needed to build libev from source
+RUN sudo yum -y install autoconf automake libtool 
+RUN wget http://dist.schmorp.de/libev/libev-4.33.tar.gz -O libev.tar.gz \
+  && tar -xvzf libev.tar.gz \
+  && cd libev-4.33 \
+  && chmod +x autogen.sh \
+  && chmod +x configure \
+  && ./autogen.sh \
+  && ./configure \
+  && make \
+  && make check \
+  && sudo make install
+
+# we really don't want to build from source, why does 'yum install gmp-devel' not work?
+# these dependencies are needed to build gmp from source
+RUN sudo yum -y install xz
+RUN wget https://gmplib.org/download/gmp/gmp-6.2.1.tar.xz -O gmp-6.2.1.tar.xz \
+  && tar -xvf gmp-6.2.1.tar.xz \
+  && cd gmp-6.2.1 \
+  && chmod +x configure \
+  && ./configure \
+  && make \
+  && make check \
+  && sudo make install 
+
+
+WORKDIR /home/opam/app
+COPY . /home/opam/app
+
+RUN repo_oxi=https://github.com/oxidizing \
+  && repo_uzh=https://github.com/uzh \
+  && opam pin add -yn sihl $repo_oxi/sihl.git \
+  && opam pin add -yn sihl-cache $repo_oxi/sihl.git \
+  && opam pin add -yn sihl-email $repo_oxi/sihl.git \
+  && opam pin add -yn sihl-queue $repo_oxi/sihl.git \
+  && opam pin add -yn sihl-storage $repo_oxi/sihl.git \ 
+  && opam pin add -yn sihl-token $repo_oxi/sihl.git \ 
+  && opam pin add -yn sihl-user $repo_oxi/sihl.git \
+  && opam pin add -yn conformist $repo_oxi/conformist.git \
+  && opam pin add -yn letters $repo_oxi/letters.git \
+  && opam pin add -yn canary $repo_uzh/canary.git \
+  && opam pin add -ywn guardian $repo_uzh/guardian.git \
```

```suggestion
  && opam pin add -yn guardian $repo_uzh/guardian.git \
```

`-w` is for working directory. We should use it without this flag.

---

this has been copied from the previous setup 🤷‍♂️